src/emacs/lean-info.el

```diff
@@ -67,6 +67,26 @@
       (lambda (&key record)
         (funcall cont record))))))
 
+(defun lean-get-info-record-at-point-sync ()
+  "Get info-record at point synchronously."
+  (with-demoted-errors "lean get info: %s"
+    (lean-server-send-synchronous-command
+     'info (list :file_name (buffer-file-name)
+                 :line (line-number-at-pos)
+                 :column (lean-line-offset)))))
+
+(defun lean-info-right-click-find-definition ()
+  "Offer to jump to definition of right-click target."
+  (interactive)
+  (let* ((info-record (plist-get (lean-get-info-record-at-point-sync) :record))
```

Comment:
> An issue with unconditionally showing "Find Definition" is that many regions don't support that. I suppose that this is also going to be an issue for holes, right?

That is true, right.  However e.g. vscode and intellij always show all go-to-definition commands, so we wouldn't be in bad company.  I guess you could even make a usability argument in *favor* of always showing go-to-definition: then the rmb-menu is predictable, and doesn't change depending on where you click.  In addition, the functionality also becomes discoverable: you can click anywhere, and get a list of all supported features.

> regexp-based heuristics

This is going to be flaky.  For example, an identifier like `generalizing` could be a constant (go-to-definition is possible), or a special token (where you can't use go-to-definition).  (If you import the standard library, it's a special token.  But it's not built-in.)

> if they don't complete in under .1 seconds, then they get ignored.

If we want to hide the menu items, then I think this is the best option.